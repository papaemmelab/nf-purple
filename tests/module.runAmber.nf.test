nextflow_process {

    name "Test Process runAmber"
    script "modules/amber.nf"
    process "runAmber"

    test("Should run Amber in matched setting") {
        when {
            params {
                tumor = "TEST_TUMOR"
                normal = "TEST_NORMAL"
                tumorBam = "${projectDir}/tests/data/input/tumor.bam"
                normalBam = "${projectDir}/tests/data/input/normal.bam"
            }
            process {
                """
                input[0] = Channel.fromPath(params.tumorBam)
                input[1] = Channel.fromPath(params.normalBam)
                """
            }
        }
        then {
            assert process.success
            assert snapshot(process.out).match()
            assert process.trace.tasks().size() == 1
            
            // check expected files
            with(process.out) {
                assert amber_baf_pcf.size() == 1
                assert amber_baf_tsv.size() == 1
                assert amber_qc.size() == 1
                assert amber_contamination_vcf.size() == 1
            }
        }
    }

    test("Should run Amber in unmatched setting") {
        when {
            params {
                tumor = "TEST_TUMOR"
                normal = null
                tumorBam = "${projectDir}/tests/data/input/tumor.bam"
                normalBam = "${projectDir}/assets/NO_FILE"
            }
            process {
                """
                input[0] = Channel.fromPath(params.tumorBam)
                input[1] = Channel.fromPath(params.normalBam)
                """
            }
        }
        then {
            assert process.success
            assert snapshot(process.out).match()
            assert process.trace.tasks().size() == 1
            
            // check expected files
            with(process.out) {
                assert amber_baf_pcf.size() == 1
                assert amber_baf_tsv.size() == 1
                assert amber_qc.size() == 1
                assert amber_contamination_vcf.size() == 0
            }
        }
    }
}
