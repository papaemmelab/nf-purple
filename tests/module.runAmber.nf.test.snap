{
    "Should run Amber in matched setting": {
        "content": [
            {
                "0": [
                    "TEST_TUMOR.amber.baf.tsv.gz:md5,582a86aee1eb90035f3a5d22cf6cdbd9"
                ],
                "1": [
                    "TEST_TUMOR.amber.baf.pcf:md5,c8d8a68055028ceea067b3c53d55aae4"
                ],
                "2": [
                    "TEST_TUMOR.amber.qc:md5,9547e68ba672e08d6bd49e1e6f654ee8"
                ],
                "3": [
                    "TEST_TUMOR.amber.contamination.vcf.gz:md5,941abd5cb8324ae2eabe82d9d0ebb6ee"
                ],
                "amber_baf_pcf": [
                    "TEST_TUMOR.amber.baf.pcf:md5,c8d8a68055028ceea067b3c53d55aae4"
                ],
                "amber_baf_tsv": [
                    "TEST_TUMOR.amber.baf.tsv.gz:md5,582a86aee1eb90035f3a5d22cf6cdbd9"
                ],
                "amber_contamination_vcf": [
                    "TEST_TUMOR.amber.contamination.vcf.gz:md5,941abd5cb8324ae2eabe82d9d0ebb6ee"
                ],
                "amber_qc": [
                    "TEST_TUMOR.amber.qc:md5,9547e68ba672e08d6bd49e1e6f654ee8"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2024-12-23T18:58:20.35175274"
    },
    "Should run Amber in unmatched setting": {
        "content": [
            {
                "0": [
                    "TEST_TUMOR.amber.baf.tsv.gz:md5,0c55ac207db6eb8e1f138a044990f4d1"
                ],
                "1": [
                    "TEST_TUMOR.amber.baf.pcf:md5,c889d8a8b2668045d35eed1fd8e2c7ce"
                ],
                "2": [
                    "TEST_TUMOR.amber.qc:md5,9547e68ba672e08d6bd49e1e6f654ee8"
                ],
                "3": [
                    
                ],
                "amber_baf_pcf": [
                    "TEST_TUMOR.amber.baf.pcf:md5,c889d8a8b2668045d35eed1fd8e2c7ce"
                ],
                "amber_baf_tsv": [
                    "TEST_TUMOR.amber.baf.tsv.gz:md5,0c55ac207db6eb8e1f138a044990f4d1"
                ],
                "amber_contamination_vcf": [
                    
                ],
                "amber_qc": [
                    "TEST_TUMOR.amber.qc:md5,9547e68ba672e08d6bd49e1e6f654ee8"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2024-12-23T18:58:34.075044669"
    }
}